nixos/release.nix: Refactoring for better multi-system support

dezgeg opened the pull request:
Currently, even if you pass `supportedSystems = [ "aarch64-linux" ]` you
end up with e.g. `nixos.iso_graphical.x86_64-linux` job. Using the new
forSubsetOfSystems avoids that.

This shouldn't affect the usual x86 trunk-combined jobset.

https://hydra.nixos.org/eval/1425558#tabs-removed for the effect on an aarch64-only jobset.

Ericson2314 commented on nixos/release.nix:
```diff
@@ -13,6 +13,8 @@ let
 
   forAllSystems = genAttrs supportedSystems;
 
+  forSubsetOfSystems = systems: genAttrs (intersectLists supportedSystems systems);
```
Should this go in release-lib.nix?

dezgeg replied:
Uff... There's apparently already an equivalent called `forAllSupportedSystems` in release-lib.nix, but nixos/release.nix is not importing release-lib.nix

Ericson2314 replied:
I'd a) import it, b) move `forAllSystems` to release-lib c) use `forAllSupportedSystems`.

dezgeg replied:
Yeah that seems the best (and `forAllSystems` could be implemented in terms of `forAllSupportedSystems` to reduce duplication). 

I hate the name `forAllSupportedSystems` though... given that we have `forAllSystems` which does something for every system passed in `supportedSystems`... and then `forAllSupportedSystems` which does something else.

Ericson2314 replied:
I have no problem with you renaming them. Maybe something like `forAllSystems` and `forTheseSystems`?

dezgeg committed:
release-lib: forAllSupportedSystems -> forTheseSystems

I'm going to move forAllSystems from nixos/release.nix, and these
functions sound too similar while doing different things.

dezgeg committed:
nixos/release.nix: Move forAllSystems to release-lib

There's already a similar forTheseSystems in release-lib, so be more
consistent.

dezgeg committed:
nixos/release.nix: Use forTheseSystems from release-lib

Currently, even if you pass `supportedSystems = [ "aarch64-linux" ]` you
end up with e.g. `nixos.iso_graphical.x86_64-linux` job. Using
forTheseSystems from release-lib avoids that.

This shouldn't affect the usual x86 trunk-combined jobset.

dezgeg commented:
Implemented the above suggestions.

Ericson2314 commented:
Glad you liked them, and thanks for making them!
